.github/workflows/workflow.yml

```diff
@@ -74,14 +74,50 @@ jobs:
     - name: Build documentation website
       run: |
         make --directory ./docs html
-       
-    - name: Publish documentation
-      uses: s0/git-publish-subdir-action@v2.2.0
-      if: contains(github.ref, 'master')
-      env:
-        REPO: self
-        BRANCH: gh-pages
-        FOLDER: docs/build/html
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+    - uses: actions/upload-artifact@v1
+      with:
+        name: docs
+        path: docs/build/html
+
+  deploy-docs:
+    needs: [tests, docs]
+    runs-on: ubuntu-latest
+    if: contains(github.ref, 'master')
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/download-artifact@v1
+        with:
+          name: docs
+          path: docs/build/html
+      - name: Publish documentation
+        uses: s0/git-publish-subdir-action@v2.2.0
+        env:
+          REPO: self
+          BRANCH: gh-pages
+          FOLDER: docs/build/html
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  deploy-docker:
+    needs: [tests, docs]
+    runs-on: ubuntu-latest
+    if: contains(github.ref, 'master')
+    steps:
+      - uses: actions/checkout@v2
+      - name: Login to gcloud registry
+        id: gcloud
+        uses: elgohr/gcloud-login-action@0.2
+        with:
+          account_key: ${{ secrets.GCLOUD_KEY }}
+      - name: Publish to Registry
+        uses: elgohr/Publish-Docker-Github-Action@2.14
```

Comment:
for the future, will this also work with dockerhub?

Reply:
yes